lib/acpi/acpi.c

```diff
@@ -345,7 +348,8 @@ int acpi_current_resource_free(ACPI_RESOURCE *res)
 }
 
 #ifdef CONFIG_PCIE_PRT
-static int acpi_get_irq_table(char *bus_name, ACPI_PCI_ROUTING_TABLE *rt_table, uint32_t rt_size)
+static int acpi_get_irq_table(char *bus_name,
+				ACPI_PCI_ROUTING_TABLE *rt_table, size_t rt_size)
```

Comment:
Please merge this entire function to the end of `acpi_legacy_irq_init()`. It's the only place where you're calling it from and it's already hard-coding the assumption that the `rt_table` parameter is always pointing at `acpi.pci_prt_table` (by iterating of the this table at the end of the function) .